Parachain pallets tests not run properly in CI

I think there is a problem in running parachain's pallet unit tests. At least for me it gives false-positive results. 

In `parachain-unit-tests` GA job there is command:
```
cargo test --locked --release -p pallet-* --lib --features=skip-ias-check
```

And once it's run on current dev, everything looks fine, tests pass.

But once I try to run tests for specific pallet like

```
cargo test --locked --release -p pallet-teerex --lib --features=skip-ias-check
```

The code cannot be even compiled:

```
kziemianek@kziemianek:~/projects/litentry/litentry-parachain$ cargo test --locked --release -p pallet-teerex --lib --features=skip-ias-check
   Compiling pallet-teerex v0.9.0 (/home/kziemianek/projects/litentry/litentry-parachain/pallets/teerex)
error[E0308]: mismatched types
   --> pallets/teerex/src/tests/skip_ias_check_tests.rs:115:4
    |
113 |         assert_ok!(Teerex::update_scheduled_enclave(
    |                    -------------------------------- arguments to this function are incorrect
114 |             RuntimeOrigin::root(),
115 |             0u32,
    |             ^^^^ expected `u64`, found `u32`
    |
note: associated function defined here
   --> pallets/teerex/src/lib.rs:476:10
    |
476 |         pub fn update_scheduled_enclave(
    |                ^^^^^^^^^^^^^^^^^^^^^^^^
477 |             origin: OriginFor<T>,
478 |             #[pallet::compact] sidechain_block_number: SidechainBlockNumber,
    |                                --------------------------------------------
help: change the type of the numeric literal from `u32` to `u64`
    |
115 |             0u64,
    |              ~~~

For more information about this error, try `rustc --explain E0308`.
error: could not compile `pallet-teerex` due to previous error
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -549,7 +549,9 @@ jobs:
         run: |
           rustup show
           cargo test --locked --release -p core-primitives --lib
-          # test pallet-teerex separately
+          # test pallet-teerex separately, with all feature combinations
+          cargo test --locked --release -p pallet-teerex --lib
+          cargo test --locked --release -p pallet-teerex --lib --features=skip-ias-check
           cargo test --locked --release -p pallet-teerex --lib --features=skip-scheduled-enclave-check
           cargo test --locked --release -p pallet-teerex --lib --features=skip-ias-check,skip-scheduled-enclave-check
           # no `skip-ias-check` feature

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -552,6 +552,7 @@ jobs:
           cargo hack test --feature-powerset --features=default --locked --release -p pallet-teerex --lib
 
       - name: Run all unittests
+        run: |
           cargo test --locked --release -p core-primitives --lib
           # no `skip-ias-check` feature
           cargo test --locked --release -p pallet-* --lib

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -549,6 +549,9 @@ jobs:
         run: |
           rustup show
           cargo test --locked --release -p core-primitives --lib
+          # test pallet-teerex separately
+          cargo test --locked --release -p pallet-teerex --lib --features=skip-scheduled-enclave-check
+          cargo test --locked --release -p pallet-teerex --lib --features=skip-ias-check,skip-scheduled-enclave-check
           # no `skip-ias-check` feature
           cargo test --locked --release -p pallet-* --lib
           # with `skip-ias-check` feature only

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -545,15 +545,14 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
-      - name: Run unittests
+      - name: Run pallet-teerex unittests
         run: |
           rustup show
+          cargo +stable install cargo-hack --locked
+          cargo hack test --feature-powerset --features=default --locked --release -p pallet-teerex --lib
+
+      - name: Run all unittests
           cargo test --locked --release -p core-primitives --lib
-          # test pallet-teerex separately, with all feature combinations
-          cargo test --locked --release -p pallet-teerex --lib
-          cargo test --locked --release -p pallet-teerex --lib --features=skip-ias-check
-          cargo test --locked --release -p pallet-teerex --lib --features=skip-scheduled-enclave-check
-          cargo test --locked --release -p pallet-teerex --lib --features=skip-ias-check,skip-scheduled-enclave-check
           # no `skip-ias-check` feature
           cargo test --locked --release -p pallet-* --lib
           # with `skip-ias-check` feature only

--- a/pallets/teerex/src/tests/mod.rs
+++ b/pallets/teerex/src/tests/mod.rs
@@ -16,5 +16,5 @@
 */
 #[cfg(feature = "skip-ias-check")]
 mod skip_ias_check_tests;
-#[cfg(not(feature = "skip-ias-check"))]
+#[cfg(all(not(feature = "skip-ias-check"), feature = "skip-scheduled-enclave-check"))]
 mod test_cases;

--- a/pallets/teerex/src/tests/skip_ias_check_tests.rs
+++ b/pallets/teerex/src/tests/skip_ias_check_tests.rs
@@ -111,7 +111,7 @@ fn register_enclave_with_empty_url_inserts_default() {
 fn register_enclave_with_scheduled_enclave_works() {
 	new_test_ext().execute_with(|| {
 		assert_ok!(Teerex::update_scheduled_enclave(
-			RuntimeOrigin::root(),
+			RuntimeOrigin::signed(AccountKeyring::Alice.to_account_id()),
 			0u64,
 			Default::default(),
 		));

--- a/pallets/teerex/src/tests/skip_ias_check_tests.rs
+++ b/pallets/teerex/src/tests/skip_ias_check_tests.rs
@@ -112,7 +112,7 @@ fn register_enclave_with_scheduled_enclave_works() {
 	new_test_ext().execute_with(|| {
 		assert_ok!(Teerex::update_scheduled_enclave(
 			RuntimeOrigin::root(),
-			0u32,
+			0u64,
 			Default::default(),
 		));
 		assert_ok!(Teerex::register_enclave(

